### ISSUE TITLE
wayland crash when drag & drop text

### ISSUE THREAD

**Issue by Reginald-Gillespie** [brave/brave-browser#37777]
Brave Version: 1.65.114 Chromium: 124.0.6367.60
Operating System: Linux 6.8.7-arch1-1

Can you reproduce this crash?
Yes

What steps will reproduce this crash? (If it's not reproducible, what were you doing just before the crash?)

I will start off by mentioning that similar issue but for images seems fixed https://github.com/brave/brave-browser/issues/36766

After updating brave a few days ago, I've not had issues with dragging images into brave like I did before. However, if I select any amount of text and drag it to any degree, brave immediately freezes and needs to be closed and reopened to work again. Arch Linux plasma 6 wayland, ozone platform parameter is set to wayland (I actually had to set this in a separate file under .config as well after Brave wouldn't start after the latest update but this is documented elsewhere).

****DO NOT CHANGE BELOW THIS LINE****
Crash ID: crash/97bb1500-3218-d60b-0000-000000000000

**Comment by lenow55** [comment]
Hello, i face with the same issue, when i try drag & drop files, texts, images brave freezes. I can only kill them and relaunch. 

**Brave version**: 1.65.114 Chromium: 124.0.6367.60
**OS**: Arch Linux x86_64 6.8.7-arch1-1

**Comment by Z-Map** [comment]
Hi, same here, the exact problematic behavior I've encountered is when I drag some type of element (mainly text selection, icons and html elements like buttons) and drop it back on the same window, all brave windows stop responding to left mouse clic event.
The ui of the window that initiated the draging event doesn't respond to anything (unable to open the menu right of the url bar even with keyboard shortcut).
The keyboard shortcut and ui on other windows still works (I can do Alt-F X to exit brave) and on restart brave say it has been closed incorrectly and allow restoring the previous session.

**Brave** : Version 1.65.122 Chromium: 124.0.6367.82 (Official Build)  (64-bit)
**OS** : Arch Linux 6.8.7-arch1-1
**DE** : Plasma 6.0.4-1, Wayland

**Comment by Xzensi** [comment]
+1

**Comment by wwboyer** [comment]
I am also having this issue on CachyOS:

**Brave:** Version 1.65.123 Chromium: 124.0.6367.91 (Official Build) (64-bit)
**OS:** CachyOS 6.8.8-2-cachyos
**DE:** Plasma 6.0.4, KWin Wayland WM

**Comment by CodeDead** [comment]
Same behavior on Fedora 40.

Brave: 1.65.126 Chromium: 124.0.6367.118 (Official Build) (64-bit) 
OS: Fedora  Linux 40 KDE Spin
DE: Plasma 6.0.4, Wayland

**Comment by Reginald-Gillespie** [comment]
To anyone who cares, it's documented more in depth [here](https://community.brave.com/t/brave-crashes-while-selecting-or-dragging-page-element-on-fedora-kde/545980/4), seems to be an issue originating in Chromium, so it's just a matter of when will brave update to the patched Chromium version.

**Comment by armando-ota** [comment]
same issue 
Operating System: KDE neon 6.0
KDE Plasma Version: 6.0.5
KDE Frameworks Version: 6.2.0
Qt Version: 6.7.0
Kernel Version: 6.5.0-35-generic (64-bit)
Graphics Platform: Wayland
Processors: 12 × Intel® Core™ i7-8086K CPU @ 4.00GHz
Memory: 62.7 GiB of RAM
Graphics Processor: NVIDIA GeForce RTX 3080 Ti/PCIe/SSE2

**Comment by Reginald-Gillespie** [comment]
Interesting, I updated and it fixed it like I mentioned but then every time after that it stopped doing it so not sure what's up with that. Maybe it was a temporary fallback to X11 instead of wayland because of some other device thing.

**Comment by armando-ota** [comment]
This is really strange ... because It cant be always reproducible :(

**Comment by Reginald-Gillespie** [comment]
For me it's constantly reproducible except for that one time that it worked fine

**Comment by SherLock707** [comment]
I’m experiencing a similar issue on Arch with Hyprland.

Brave Browser Version: 1.69.162

Whenever I select and drag content (text, link, image) within any website, the browser freezes.

**Comment by Nintron27** [comment]
Having the same issue on NixOS 24.05 with Hyprland. Any selecting of text and dragging, or any dragging of links or images freezes the browser. I can escape the freeze by Ctrl+W and then reopening the tab. Changing the ozone platform to X11 or Wayland makes no difference.

Brave Browser Version: 1.69.168

**Comment by Twikzer** [comment]
this still hapens today, this is ridiculous

Brave [v 1.77.100]
[1.77.100 Chromium: 135.0.7049.100 (Build oficial) (64 bits)]
Arch Linux 6.14.2
Kde Plasma 6.3.4
Nvidia 570.133.07

**Comment by CodeDead** [comment]
> this still hapens today, this is ridiculous
> 
> Brave [v 1.77.100]
> [1.77.100 Chromium: 135.0.7049.100 (Build oficial) (64 bits)]
> Arch Linux 6.14.2
> Kde Plasma 6.3.4
> Nvidia 570.133.07

Yup. I gave up and switched to Firefox. Haven't looked back since. Chromium browsers on Linux are just a pain to keep running. If it's not video decoding that breaks, its drag and drop that breaks or notifications or journalctl spam with all kinds of warnings filling up your drives etc...

**Comment by Revival8697** [comment]
Are everyone here running Nvidia?

**Comment by Reginald-Gillespie** [comment]
> Is everyone here running Nvidia?

Originally when I posted I was running an intel iGPU with an nvidia eGPU, so I did have the drivers installed but I believe the issue occurred even when that GPU was not active. I could be wrong, it's been a while and I'm on a new computer now.

**Comment by Syntist** [comment]
Same having this issue, using IGPU.

**Comment by ajama-1984** [comment]
> Is everyone here running Nvidia?

No, Got an AMD GPU and same issue.
